components/Tabs/Tabs.js

```diff
@@ -1,61 +1,64 @@
 class TabLink {
   constructor(tabElement){
     // assign this.tabElement to the tabElement DOM reference
-    // this.tabElement;
+    this.tabElement = tabElement;
     
     // Get the `data-tab` value from this.tabElement and store it here
-    // this.tabData = ; 
+    this.tabData = this.tabElement.dataset.tab; 
     
     // We need to find out if a user clicked 'all' cards or a specific category.  Follow the instructions below to accomplish this task:    
     
-    /* <- Delete this comment block when you work on the if statement
+  
     // Check to see if this.tabData is equal to 'all'
-    if(){
+ 
+      if(this.tabData === "all") {
       // If `all` is true, select all cards regardless of their data attribute values
-      // this.cards = ;
+      this.cards = document.querySelectorAll('.card');
     } else {
       // else if `all` is false, only select the cards with matching this.tabData values
-      // this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);
     }
-    /* <- Delete this comment block when you work on the if statement
 
-     // Map over the newly converted NodeList we just created in our if statement above. Convert each this.cards element into a new instance of the TabCard class. Pass in a card object to the TabCard class. 
-    // this.cards = Array.from(this.cards).map();
+     // Map over the newly converted NodeList we just created in our if statement above. 
+    //  Convert each this.cards element into a new instance of the TabCard class. 
+    //  Pass in a card object to the TabCard class. 
+      this.cards = Array.from(this.cards).map(x => new TabCard(x));
```

Comment:
This is correct just try to be more specific when you use your map so is easier to understand for people reading your code.

````
.map(card => new TabCard(card))
````

Reply:
Ok thank you Marguel!